LAVA_setup_guide.md

```diff
@@ -0,0 +1,192 @@
+- you may now login as user
+- `wget https://apt.lavasoftware.org/lavasoftware.key.asc`
+- `sudo apt-key add lavasoftware.key.asc`
+- `sudo apt-get update`
+
+Note: although it is not officially supported, lava-server can be installed in Ubuntu (e.g. on WSL) by adding the below step:
+`sudo apt-get install postgresql && sudo service postgresql start`
+
+This resolves a dpkg-configure error that occurs because Postgres is not listening to port 5432 during installation
+
+### Server setup
+
+Here is the process you'll need to build the server from scratch. It's recommended you use the docker compose instead -
+see below. 
+
+#### Install LAVA server
+
+- `sudo apt install lava-server`
+- `sudo a2dissite 000-default`
+- `sudo a2enmod proxy`
+- `sudo a2enmod proxy_http`
+- `sudo a2ensite lava-server.conf`
+- `sudo service apache2 restart`
+
+#### Setup user
+
+- `sudo lava-server manage users add <usernane> --passwd <password>`
+- `sudo lava-server manage authorize_superuser --username <usernane>`
+
+#### Set website URL (this affect internal web interface links)
+
+- go to http://localhost/admin/sites/site/
+- click on example.com and delete it
+- add new one with the correct url
+- edit /usr/lib/python3/dist-packages/lava_server/settings/common.py
+    * search for SITE_ID and change it to SITE_ID = 2
+    * `sudo service lava-server-gunicorn restart`
+    * any time you change site domain you have to change the index to match
+
+#### Add Worker
+
+- `sudo lava-server manage workers add <WORKER_NAME>`
+    * don't do that on a single machine install, it will break lava
+    * set the hostname to the worker name
+- copy the token string to be used when configuring the worker
+
+#### Add device definition
+
+- create a new file in /etc/lava-server/dispatcher-config/devices with the name of the device (you can use the one from https://github.com/ARMmbed/mbed-os-lava-scripts)
+    * You can just copy the whole directory https://github.com/ARMmbed/mbed-os-lava-scripts/tree/master/lava-server to /etc/, otherwise follow steps below
+    * must have extension jinja2: `mbed.jinja2`
+    * the device should inherit from a device type in /usr/share/lava-server/device-types, to inherit from docker add line: `{% extends "docker.jinja2" %}`
+    * Must pass in extra parameters to docker: `{% set docker_extra_arguments = ["-v /dev:/dev --group-add=dialout --device-cgroup-rule 'a 166:* rwm'"] %}`
+- `sudo service lava-server-gunicorn restart`
+
+### Worker setup
+
+#### Install dependencies
+
+`sudo apt install docker.io git`
+Install docker following official [docker documentation](https://docs.docker.com/engine/install/debian/).
+`sudo apt install bluez bluetooth`
+`sudo killall -9 bluetoothd`
+    * daemon cannot be running when docker starts
+    * you can change AutoEnable=true to false in /etc/bluetooth/main.conf but it still needs to start at least once
+
+#### Install worker
+
+`sudo apt install lava-dispatcher`
```

Comment:
Lava requires to use the same version for its server and connected workers.
However, it's possible that this requirement will be not fulfilled because using `https://apt.lavasoftware.org/release buster ` repository will always give us the newest possible stable version when in the same time lava-server configured by `mbed-os-lava-docker-compose/.env` file is fixed to 2021.03.post1 version (so adding new workers in the future will be not possible due to this version mismatched)

You can mention here that older version of lava-dispatcher/server can be installed from snapshots repository:
`https://apt.lavasoftware.org/snapshot/`
More information about versioning and releasing new version can be found here: `https://docs.lavasoftware.org/lava/installing_on_debian.html `

Reply:
thanks, I will add an explanation you cannot mix versions